lib/fluent/plugin/filter_kubernetes_metadata.rb

```diff
@@ -161,6 +161,20 @@ def initialize
       @prev_time = Time.now
     end
 
+    def periodically_read_bearer_token_file
+      token_read_period = 10
+      while true
+        log.debug "Sleeping for #{token_read_period}"
+        sleep(token_read_period)
+        log.debug "Updating bearer_token"
+        # This will be enough with kubeclient 5
+        # @client.instance_variable_set(:@faraday_client, nil)
+        # until then
+        @client.bearer_token(File.read(@bearer_token_file))
+        # @client.instance_variable_set(:@rest_client, nil)
```

Comment:
All these options rely on knowing the internals of kubeclient.  
How hard it is to recreate the whole `Kubeclient::Client.new(...)`?  I see that's what #326 was going for :+1:

Also both `@rest_client` (kubeclient 4) and `@faraday_client` (5) only cover CRUD actions but not watches, doesn't this plugin rely on watches?

- I don't think an open watch socket will benefit from any of these approaches, but OTOH it's already open and doesn't need auth, right?  At some point server might disconnect it and you'll open new watch with fresh auth.

I guess that's why you closed this and helped improve the file re-reading in kubeclient itself. (Thanks for that!)  

But I'm afraid I'm not yet close to releasing kubeclient 5.  I'm working through a backlog in my spare time but it might take a month at least (and that's not a commitment!)
If you need this soon, I recommend the #326 approach.

Reply:
@cben Thanks for sharing this.
I thought recreating the client in #326 was a bit brutal (and I had trouble with the failing tests ...) but I agree with the fact that relying on kubeclient internals may not be the best way (and it still doesn't work actually).
I think I kinda fixed the tests in #326.
This PR and #326 share the same periodic approach, I'll wait until @jcantrill answers https://github.com/fabric8io/fluent-plugin-kubernetes_metadata_filter/pull/327#discussion_r828193326 to close this one and re-open #326